common/luaclass.c

```diff
@@ -300,6 +300,27 @@ luaA_class_connect_signal_from_stack(lua_State *L, lua_class_t *lua_class,
                                      const char *name, int ud)
 {
     luaA_checkfunction(L, ud);
+
+    /* Duplicate the function in the stack */
+    lua_pushvalue(L, ud);
+    luaA_checkfunction(L, -1);
+    luaA_checkfunction(L, ud);
+
+    /* Create a new signal to notify there is a global connection. */
+    char buf[128];
+    a_strcpy(buf, 128, name);
+    a_strcat(buf, 128, "::connected");
+
+    /* Emit a signal to notify Lua of the global connection.
+     *
+     * This can useful during initialization where the signal needs to be
+     * artificially emitted for existing objects as soon as something connects
+     * to it
+     */
+    luaA_checkfunction(L, -1);
```

Comment:
Okay, here is the fourth duplicate. This code is really making sure that it gets a function...

Reply:
done